wp-sw-cache/wp-sw-cache-main.php

```diff
@@ -21,18 +22,36 @@ public static function init() {
     }
   }
 
+  public function update_version($name = '') {
+    if(!$name) {
+      $name = time();
+    }
+    update_option('wp_sw_cache_name', self::$cache_prefix.'-'.$name);
+  }
+
   public function write_sw() {
+
     $files = get_option('wp_sw_cache_files');
+    $file_keys = array();
     if(!$files) {
-        $files = array();
+      $files = array();
     }
     foreach($files as $index=>$file) {
-        $files[$index] = get_template_directory_uri().'/'.$file;
+      $tfile = get_template_directory().'/'.$file;
+
+      if(file_exists($tfile)) {
+        $file_keys[get_template_directory_uri().'/'.$file] = filemtime($tfile);
+      }
     }
 
+    $file_keys = array_keys($file_keys);
+    $name = md5(serialize($file_keys));
```

Comment:
Thinking out loud, and I'm very tired so bear with me, the reason I did an md5 on a serialized array of files and times is that the time may not be the only thing that changes -- the file listing could change (via admin) as well.

And since we need to verify a file exists before outputting it to the SW (even if they chose the file in admin, they may have deleted the file manually from the server), we need o do all of these checks in the outputting SW function